packages/api/src/post-build.js

```diff
@@ -105,24 +106,18 @@ class PostBuildHandler {
   }
 
   listen () {
-    // route for when parsed in action
-    this.app.post('/api/buildparsed', async (req, res, next) => {
-      try {
-        var buildInfo = this.parseBuildInfo(req.body.metadata);
-        var testCases = this.parseTestCases(req.body.data);
-        await addBuild(testCases, buildInfo, this.client, global.headCollection);
-        res.send({ message: 'successfully added build' });
-      } catch (err) {
-        handleError(res, err);
-      }
-    });
-
     // route for parsing test input server side
     this.app.post('/api/build', async (req, res, next) => {
       try {
-        var buildInfo = this.parseBuildInfo(req.body.metadata);
-        var parsedRaw = await this.parseRawOutput(req.body.data, req.body.type);
-        var testCases = this.parseTestCases(parsedRaw);
+        const buildInfo = this.parseBuildInfo(req.body.metadata);
+        const parsedRaw = await this.parseRawOutput(req.body.data, req.body.type);
+        const testCases = this.parseTestCases(parsedRaw);
+
+        const isValid = await validateGithub(req.body.metadata.github.token, req.body.metadata.github.repository);
```

Comment:
Once you've used the token to check validity, how long do you keep it? Are we storing it in the database as part of the metadata?

Reply:
It is never stored. An api request is made using it and that is all.